### ISSUE TITLE
The date template function doesn't support unix epoch values

### ISSUE THREAD

**Issue by MichelHollands** [grafana/loki#8624]
**Is your feature request related to a problem? Please describe.**

Using the `date` template function to parse fields with epoch times doesn't work. Consider for example log line like `{"from":1618492542349,"to":1618492842349,"message":"some message"}`. Running a LogQL query like `json | __error__ = "" | line_format '{{ date "2006-01-02T15:04:05.00" .from}}'` won't give the expected result. The from field is a label and therefore string. That is not supported as input and defaults to the current date and time. See the sprig library [here](https://github.com/Masterminds/sprig/blob/master/date.go#L14).   

**Describe the solution you'd like**
The `date` template function should support string arguments which can be epoch times (or dates).

**Describe alternatives you've considered**
The current workaround is as follows, which is not very user friendly.

```
| line_format "{{ date `2006-01-02T15:04:05.00` (div (int .from) 1000) }}"
```

**Comment by ashwanthgoli** [comment]
wdyt about having `toDate` handle epoch times? it could support `sec, ms, ns` in the format string.
```
{{ toDate "sec" "1677480118" }}
```

> Or perhaps the date function could error when given string input.

Agree with this as it's returning `time.Now()` in the [default](https://github.com/Masterminds/sprig/blob/581758eb7d96ae4d113649668fa96acc74d46e7f/date.go#L29) case which is misleading for users.

**Comment by eisandbar** [comment]
Is something like #8774 what you wanted? @MichelHollands

**Comment by MichelHollands** [comment]
@eisandbar No, because that converts from a date to a date in another timezone or converts a date to unix epoch. 
The requested functionality is from the ms and ns epoch times to a date.
